### PR TITLE
Make gradient clipping configurable.

### PR DESCRIPTION
**Pull request by Tixxx**
add control flag to c++ and python frontend

**Description**:
make gradient clipping configurable

**Motivation and Context**
provide users freedom to enable/disable gradient clipping for more efficient experimentation.

### CODE REVIEW

**Comment by kit1980** [comment]
Why it's just a boolean to enable/disable?
Shouldn't it be a value to clip, and also if clip value or norm, and which norm, like `clip_grad_norm_` and `clip_grad_value_` in PyTorch?

**Comment by Tixxx** [comment]
> Why it's just a boolean to enable/disable?
> Shouldn't it be a value to clip, and also if clip value or norm, and which norm, like `clip_grad_norm_` and `clip_grad_value_` in PyTorch?

@kit1980  other clip values haven't been tested, we are just using the default value used in most training scenarios so likely wont expose that right now. 
This is clipping norm, would changing this to enable_grad_norm_clip work for you?

**Comment by kit1980** [comment]
> Why it's just a boolean to enable/disable?
> Shouldn't it be a value to clip, and also if clip value or norm, and which norm, like clip_grad_norm_ and clip_grad_value_ in PyTorch?
> 
> @kit1980  other clip values haven't been tested, we are just using the default value used in most training scenarios so likely wont expose that right now.
> This is clipping norm, would changing this to enable_grad_norm_clip work for you?

I think it's o for now. Maybe add a todo comment?

File: orttraining/orttraining/models/runner/training_runner.h [hunk 1]
```diff
@@ -149,6 +149,8 @@ class TrainingRunner {
 
     int data_parallel_size = 1;
     int horizontal_parallel_size = 1;
+    // Enable gradient clipping.
+    bool enable_gradient_clip=true;
```
Comment (SherlockNoMad) [inline orttraining/orttraining/models/runner/training_runner.h]: >true [](start = 30, length = 4)

What's our current default behavior for gradient clipping? 

Is it on ?

Comment (Tixxx) [reply orttraining/orttraining/models/runner/training_runner.h]: it's got different hard-coded behaviors in different optimizer builders. For Adam. default is off, for lamb is on..  It's better to have a control knob for this.

Comment (SherlockNoMad) [reply orttraining/orttraining/models/runner/training_runner.h]: Right... So our Adam's default behavior would be different with this change 

Please let the team know... 

---
In reply to: [395801750](https://github.com/microsoft/onnxruntime/pull/3243#discussion_r395801750) [](ancestors = 395801750)
